src/Modal/Modal.tsx

```diff
@@ -108,8 +108,10 @@ class Modal extends React.Component<ModalProps, ModalState> {
       bodyStyles: React.CSSProperties;
     } = {
       modalStyles: {
-        paddingRight: bodyIsOverflowing && !modalIsOverflowing ? getScrollbarSize() : 0,
-        paddingLeft: !bodyIsOverflowing && modalIsOverflowing ? getScrollbarSize() : 0
+        [doc.dir !== 'rtl' ? 'paddingRight' : 'paddingLeft']:
+          bodyIsOverflowing && !modalIsOverflowing ? getScrollbarSize() : 0,
+        [doc.dir !== 'rtl' ? 'paddingLeft' : 'paddingRight']:
+          !bodyIsOverflowing && modalIsOverflowing ? getScrollbarSize() : 0
       },
```

Comment:
In actual development, some development engineers will set the rtl property on the body tag.  So,
```js
(document.body.getAttribute('dir') || doc.dir) !=='rtl'
```
Or, add an `isRTL` method.
```js
function isRTL() {
  return (
    typeof window !== 'undefined' && (document.body.getAttribute('dir') || document.dir) === 'rtl'
  );
}
```

Reply:
@simonguo The recommendation is to add `dir="rtl"` to the html tag. See: 
* https://www.w3.org/International/questions/qa-html-dir
* https://www.w3.org/International/questions/qa-bidi-css-markup 

Anyway, we can also read the effective direction:

```js
typeof window !== 'undefined' && window.getComputedStyle(document.body).direction === 'rtl';
```

Which gives the freedom of setting the direction via `dir` attribute or `direction` style on either `body` or `html` node.